Sources/Tools/MaxMain/MaxCompat.h

```diff
@@ -129,6 +129,39 @@ You can contact Cyan Worlds, Inc. by email legal@cyan.com
 #   define MAX_REF_PROPAGATE_VALUE propagate
 #endif
 
+#if MAX_VERSION_MAJOR < 24 // Max 2022
+#   define MAX_NAME_LOCALIZED1
+#   define MAX_NAME_LOCALIZED2
+#   define MAX_NAME_LOCALIZED_DEFAULT
+#   define MAX_NAME_LOCALIZED_VALUE
+#   define MAX24_CONST
+
+#   define plClassDesc2 ClassDesc2
+#   define plClassDesc ClassDesc
+
+#   define GetSystemUnitInfo GetMasterUnitInfo
+#else
+#   define MAX_NAME_LOCALIZED1 bool localized
+#   define MAX_NAME_LOCALIZED2 , bool localized
+#   define MAX_NAME_LOCALIZED_DEFAULT = true
+#   define MAX_NAME_LOCALIZED_VALUE , localized
+#   define MAX24_CONST const
+
+#   include <iparamb2.h> // Dammit
+
+class plClassDesc2 : public ClassDesc2
+{
+public:
+    const MCHAR* NonLocalizedClassName() override { return ClassName(); }
+};
+
+class plClassDesc : public ClassDesc
+{
+public:
+    const MCHAR* NonLocalizedClassName() override { return ClassName(); }
+};
+#endif
```

Comment:
I wonder if this same pattern can be extended to some of the other nonsense...  For example, instead of having the MAX_NAME_LOCALIZED* macros above, just implement a base class which forwards the flavor with the `localized` parameter to a pure virtual method matching the old signature (which is of course already overridden everywhere)

Reply:
I was initially thinking the same thing but disregarded it because it would probably require multiple inheritance... The class hierarchy is significantly more complicated for the ones that use the NAME_LOCALIZED junk. Any preference?

Reply:
Ok, I experimented with changing these all to proxy methods. Unfortately, I wasn't really able to come up with a workable solution. In some cases, Autodesk left the old method signature behind but sealed it so overloading is now an error... eg with `GetObjectName()` and `GetClassName()`. The sealed variant automatically proxies to the new localized one. In other cases, (eg the Sub/Slot/Mtl Name stuff) the method simply gained the `bool localized` with a default argument. I'm really at a loss as to how to make this less terrible.